src/skipmissing.jl

```diff
@@ -35,10 +35,12 @@ end
 
 _missing(x, itr) = ismissing(x) || x == missingval(itr)  # mind the order, as comparison with missing returns missing
```

Comment:
I think we need the check on this method as well:
```suggestion
 function _missing(x, itr) 
    # mind the order, as comparison with missing returns missing
    if isnothing(missingval(itr))
        ismissing(x)
    else
        ismissing(x) || x == missingval(itr)
    end
end
```

Reply:
I think it should be fine as it is since `ismissing(nothing)` returns `false`, so there's no need for the first clause.

```julia
julia> ismissing(nothing)
false
```

Unless you want `skipmissing` to keep `nothing` values when `missingval=nothing`?

Reply:
Hmm hard to know whats correct there. 

Myvidea was that `missingval` being `nothing` means there is no `missingval` behaviour at all, rather than the missing value is `nothing`.

Reply:
It's a strange edge case. It seems there are three reasonable solutions:
1) Treat `nothing` like `missing` and remove it always.
2) Remove `nothing` only when `missingval = nothing`.
3) Never remove `nothing`.

Reply:
If `isnothing(missingval)` we just don't set any `nodata` value in gdal:
 
https://github.com/rafaqz/Rasters.jl/blob/main/ext/RastersArchGDALExt/gdal_source.jl#L516-L531

It represents the case where there is no missing data in the raster.

So `nothing` being seen as a `missingval` could be inconsistent with this. Probably we should not remove `nothing` here either, if that also makes sense to you.

Reply:
Sure, I'll modify the code accordingly. If everything looks good to you after that, I think we can close this PR.